pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -430,6 +430,11 @@ func (c *Cloud) CreateMasterPool(p model.MasterPool) error {
 		return err
 	}
 
+	validMachine := validateMachineType(p.MachineType)
```

Comment:
Is it worthwhile pulling this check back out a level to the `create.go:validateCreateFlags` function? It's effectively validation of the flags being provided, and then can be defined just once with different implementations of the `validateMachineType` function for different providers.